idx-kotlin/src/main/java/com/okta/idx/kotlin/client/InteractionCodeFlow.kt

```diff
@@ -145,9 +147,19 @@ class InteractionCodeFlow internal constructor(
      * Evaluates the given redirect url to determine what next steps can be performed. This is usually used when receiving a redirection from an IDP authentication flow.
      */
     suspend fun evaluateRedirectUri(uri: Uri): IdxRedirectResult {
-        if (!uri.toString().startsWith(flowContext.redirectUrl)) {
-            val error = "IDP redirect failed due not matching the configured redirect uri."
-            return IdxRedirectResult.Error(error)
+        val mismatchedUriError = IdxRedirectResult.Error(
+            "IDP redirect failed due not matching the configured redirect uri."
+        )
+        val otpQueryParameter = uri.getQueryParameter("otp")
+        if (!uri.toString().startsWith(flowContext.redirectUrl) && otpQueryParameter == null) {
+            return mismatchedUriError
+        }
+        if (otpQueryParameter != null) {
+            if ((flowContext.emailRedirectUrl == null) ||
+                !uri.toString().startsWith(flowContext.emailRedirectUrl)
+            ) {
+                return mismatchedUriError
+            }
```

Comment:
Hmm my Kotlin experience isn't very great, but I wonder if there's a way to clean up / refactor this method to make it clearer what's going on?  I find it difficult to follow the logic with the collection of nested statements.

For context, this is the equivalent function in the Swift implementation. https://github.com/okta/okta-idx-swift/blob/f8080641556af44355b2003fab12a6d7c55b6f86/Sources/OktaIdx/InteractionCodeFlow.swift#L234

Do you think there's some way to improve the legibility of this @rajdeepnanua-okta?

Additionally, in the Swift example, we didn't need to add OTP redirect handling to the Swift RedirectResult enum since this behaviour will largely be controlled by the developer. So I don't know if adding otpQuery handling to this function makes sense.

Reply:
I've removed all the changes inside SDK and limited EML functionality to just the sample app. I agree, this method is doing a bit too much right now and it would be good to refactor it.